openedx/core/djangoapps/cache_toolbox/middleware.py

```diff
@@ -144,24 +145,27 @@ def _verify_session_auth(self, request):
             session_hash = request.session.get(HASH_SESSION_KEY)
             session_hash_verified = session_hash and constant_time_compare(
                 session_hash, request.user.get_session_auth_hash())
+            
+            # session hash is verified from the default algo, so skip legacy check
             if session_hash_verified:
-                set_custom_attribute(
-                    'CacheBackedAuthenticationMiddleware.verify.alg', settings.DEFAULT_HASHING_ALGORITHM
+                set_custom_attribute('session_hash_verified', True)                
+                return
+            
+            if (
+                session_hash and
+                hasattr(request.user, '_legacy_get_session_auth_hash') and
+                constant_time_compare(
+                    session_hash,
+                    request.user._legacy_get_session_auth_hash()  # pylint: disable=protected-access
                 )
-            else:
-                if not (
-                    session_hash and
-                    hasattr(request.user, '_legacy_get_session_auth_hash') and
-                    # pylint: disable: protected-access
-                    constant_time_compare(
-                        session_hash,
-                        request.user._legacy_get_session_auth_hash()  # pylint: disable=protected-access
-                    )
-                ):
-                    # The session hash has changed due to a password
-                    # change. Log the user out.
-                    request.session.flush()
-                    request.user = AnonymousUser()
-                    _mark_cookie_for_deletion(request)
-                else:
-                    set_custom_attribute('CacheBackedAuthenticationMiddleware.verify.alg', "sha1")
+            ):
+                # session hash is verified from legacy hashing algorithm.
+                set_custom_attribute('legacy_session_hash_verified', True)
```

Comment:
[Nit] Might be simpler to use of you rename this to `session_hash_verified` and make the value "fallback" (or "legacy"), and change it to "default" (from True) above.